### PR TITLE
tests/gnrc_udp: Replace atoi() by strtol().

### PR DESCRIPTION
**Pull request by jcarrano**
### Contribution description

atoi() cannot detect errors. Many implementation return zero on error and that is what was being checked here, making the "udp send" command unable to parse integer values of zero. On top of this, the behavior on errors does not seem to be specified in the standard (so it is not even correct to check for zero even when zero is not an accepted value, like for a port number).

The result of all this is that sending UDP packets of zero length (as required by the Release Specs) was not possible.

This patch replaces atoi by strlen, which allows for robust error detection. Sending zero length packets is possible.

### Testing procedure

Run `make -C tests/gnrc_udp all term`

In the RIOT terminal, try to send a 0 length packet:

```
> udp send fe80::bd:b7e 1337 0
udp send fe80::bd:b7e 1337 0
Error: unable to parse data_len
```

With this patch:

```
> udp send fe80::bd:b7e 1337 0
udp send fe80::bd:b7e 1337 0
Success: send 0 byte to [fe80::bd:b7e]:1337
```

### Issues/PRs references

None.

### CODE REVIEW

File: tests/gnrc_udp/udp.c [hunk 1]
```diff
@@ -208,19 +221,27 @@ int udp_cmd(int argc, char **argv)
     }
 
     if (strcmp(argv[1], "send") == 0) {
+        char *conversion_end;
         uint32_t num = 1;
         uint32_t delay = 1000000LU;
+        if (argc > 5) {
+            num = strtol(argv[5], &conversion_end, 0);
+            if (*conversion_end != '\0') {
+                goto send_syntax_err;
```
Comment (miri64) [inline tests/gnrc_udp/udp.c]: Rather use a `_usage()` function here and just `return 1`. Jumping into the middle of some `if`-handling just invites spaghetti code.

File: tests/gnrc_udp/udp.c [hunk 2]
```diff
@@ -97,14 +109,14 @@ static void send(char *addr_str, char *port_str, char *data_len_str, unsigned in
...
+    port = valid_port(strtoul(port_str, &conversion_end, 0));
+    if (*conversion_end != '\0' && port != 0) {
         puts("Error: unable to parse destination port");
         return;
     }
 
-    data_len = atoi(data_len_str);
-    if (data_len == 0) {
+    data_len = strtoul(data_len_str, &conversion_end, 0);
+    if (*conversion_end != '\0') {
```
Comment (miri64) [inline tests/gnrc_udp/udp.c]: Sleeping over it, let's just focus on fixing this one for now for the release. The other stuff should also be fixed in applications like [`examples/gnrc_networking`](https://github.com/RIOT-OS/RIOT/blob/7484af6d369a6af44671e1564ee8457b8f033aba/examples/gnrc_networking/udp.c)

**Comment by miri64** [comment]
Ping @jcarrano?

**Comment by jcarrano** [comment]
@miri64 I stripped down the patch.

**Comment by miri64** [comment]
Murdocks complaining about an unrelated error here https://github.com/RIOT-OS/RIOT/blob/e0de94cb5f247e7c44ad24ef3a6b642ca923199d/tests/gnrc_udp/udp.c#L62. Can you please piggy-back a fix?

**Comment by miri64** [comment]
Please adapt your backport.

**Pull request merged**